helix-term/src/commands.rs

```diff
@@ -2315,10 +2315,25 @@ enum Operation {
     Change,
 }
 
+fn only_whole_lines(selection: &Selection, text: &Rope) -> bool {
+    selection.ranges().iter().all(|range| {
+        // If not at least a full line is selected (strange require 2).
+        if range.slice(text.slice(..)).len_lines() < 2 {
+            return false;
+        }
```

Comment:
The "strange require 2" part is a bit confusing here in the comment - is that just commentary on ropey's len_lines?

To clarify about ropey's behavior: `len_lines` returns the count of line breaks in the slice + 1, so even for an empty slice/rope `len_lines` will return `1`. If there are any line breaks then this will be 2 or more.